### PR TITLE
vc-build 1.7.2

### PR DESCRIPTION
**Pull request by krankenbro**

### CODE REVIEW

**Comment by mvktsk** [comment]
Review task https://virtocommerce.atlassian.net/browse/VP-7391 has been created

File: build/Build.cs [hunk 1]
```diff
@@ -815,7 +815,11 @@ private async Task<string> SendSwaggerSchemaToValidator(HttpClient httpClient, s
 
             var processEnd = ProcessTasks.StartProcess(dotNetPath, endCmd, customLogger: SonarLogger, logInvocation: false)
                 .AssertWaitForExit().AssertZeroExitCode();
-            processEnd.Output.EnsureOnlyStd();
+            var errors = processEnd.Output.Where(o => !o.Text.Contains(@"The 'files' list in config file 'tsconfig.json' is empty") && o.Type == OutputType.Err);
```
Comment (Vectorfield4) [inline build/Build.cs:818]: Prevent double enumeration.
```suggestion
            var errors = processEnd.Output.Where(o => !o.Text.Contains(@"The 'files' list in config file 'tsconfig.json' is empty") && o.Type == OutputType.Err).ToList();
```

File: build/PlatformTools/AppSettingsExtension.cs [hunk 2]
```diff
@@ -9,8 +9,12 @@ public static class AppSettingsExtension
...
         public static string GetModulesDiscoveryPath(this IConfiguration configuration)
         {
+            var defaultDiscoveryPath = "./modules";
+            if (configuration == null)
+                return defaultDiscoveryPath;
+
             var virtoSection = configuration.GetSection("VirtoCommerce");
-            var result = virtoSection.GetValue<string>("DiscoveryPath", "./modules");
+            var result = virtoSection.GetValue<string>("DiscoveryPath", defaultDiscoveryPath);
             return result;
```
Comment (Vectorfield4) [inline build/PlatformTools/AppSettingsExtension.cs:18]:
```suggestion
            return result ?? defaultDiscoveryPath ;
```

File: build/PlatformTools/Build.PackageManager.cs [hunk 3]
```diff
@@ -104,39 +151,75 @@ private bool NeedToInstallPlatform(string version)
...
+        moduleInstaller.Install(modulesToInstall.Where(m => !m.IsInstalled), progress);
         localModuleCatalog.Reload();
     });
 
+    private bool ModuleInstalled(string discoveryPath, string moduleId, string version)
+    {
+        var result = false;
+        var moduleDir = Path.Combine(discoveryPath, moduleId);
+
+        return result;
```
Comment (Vectorfield4) [inline build/PlatformTools/Build.PackageManager.cs:220]: `result` is always false.

File: build/PlatformTools/GithubManager.cs [hunk 4]
```diff
@@ -16,10 +16,27 @@ class GithubManager
...
         }
+        public static async Task<Release> GetPlatformRelease(string token, string releaseTag)
+        {
+            SetAuthToken(token);
+            return await GetPlatformRelease(releaseTag);
+        }
+        public static void SetAuthToken(string token)
+        {
+            if(!string.IsNullOrEmpty(token))
+                client.Credentials = new Credentials(token);
```
Comment (Vectorfield4) [inline build/PlatformTools/GithubManager.cs:31]:
```suggestion
                Client.Credentials = new Credentials(token);
```

File: build/PlatformTools/PackageManager.cs [hunk 5]
```diff
@@ -23,6 +23,11 @@ public static PackageManifest CreatePackageManifest(string platformVersion, stri
             return manifest;
         }
 
+        public static PackageManifest CreatePackageManifest(string platformVersion)
```
Comment (Vectorfield4) [inline build/PlatformTools/PackageManager.cs:26]:
```suggestion
        public static PackageManifest BuildDefaultPackageManifest(string platformVersion)
```

**Review by Vectorfield4 — CHANGES_REQUESTED** [review]

**Pull request merged**